src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -1063,82 +1068,95 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
     }
 
     private async handleConnectionErrorCodes(
-        result: ConnectionCompleteParams,
+        result: ConnectionInfo,
         state: ConnectionDialogWebviewState,
     ): Promise<ConnectionDialogWebviewState> {
-        if (result.errorNumber === errorSSLCertificateValidationFailed) {
-            this.state.connectionStatus = ApiStatus.Error;
-            this.state.dialog = {
-                type: "trustServerCert",
-                message: result.errorMessage,
-            } as TrustServerCertDialogProps;
+        const errorType = await getSqlConnectionErrorType(
+            {
+                errorNumber: result.errorNumber,
+                errorMessage: result.errorMessage,
+                message: result.messages,
+            },
+            result.credentials,
+        );
+        switch (errorType) {
```

Comment:
I don't love changing this to a switch statement, unless we're also going to shift the logic for each one into a helper method.  Larger case statements make it easier to miss something falling through or incorrect scoping.